### PR TITLE
UI: Show a loading message during shader preload

### PR DESCRIPTION
**Pull request by unknownbrackets**
A few things:
 * Add a UI tween class to manage animations (didn't change anywhere yet.)
 * Make shader precompile asynchronous, but still block game start.
 * Show a "Loading game..." message while precompiling, which fades out on top of game (but only shows if loading takes time.)

Also, with this, we could for example also preload shaders on Vulkan, but probably do it just using a thread.

Not sure if this might've been causing ANRs or something on first load when GL took several seconds to preload.

-[Unknown]

### CODE REVIEW

File: GPU/GLES/ShaderManagerGLES.h [hunk 1]
```diff
@@ -167,7 +167,8 @@ class ShaderManagerGLES : public ShaderManagerCommon {
 	std::vector<std::string> DebugGetShaderIDs(DebugShaderType type);
 	std::string DebugGetShaderString(std::string id, DebugShaderType type, DebugShaderStringType stringType);
 
-	void LoadAndPrecompile(const std::string &filename);
+	void Load(const std::string &filename);
+	bool ContinuePrecompile(float sliceTime = 1.0f / 60.0f);
```
Comment (unknownbrackets) [inline GPU/GLES/ShaderManagerGLES.h]: Maybe this should be slightly less, in case we would lose a frame to vsync?

-[Unknown]

**Comment by hrydgard** [comment]
Neat, gonna check it out in a bit.

For Vulkan we can do even better than compiling on a thread - it's possible to dump the VkPipelineCache object to disk, and once reloaded (which should be very fast) recreation of any pipelines that are in it should be instant.

**Comment by unknownbrackets** [comment]
Cool, although I wonder if some drivers will still take a little time (probably much less than GL, in any case.)

-[Unknown]

**Comment by hrydgard** [comment]
In that case we can always load the dump and then immediately do the GL thing where we precreate everything. And yeah, we could do that on a thread I suppose.

File: GPU/GLES/ShaderManagerGLES.h (hunk 1 above)
Comment (hrydgard) [reply GPU/GLES/ShaderManagerGLES.h]: That will already be rounded very slightly down due to fp precision :)  But let's see how it works.

**Comment by marosis** [comment]
Really nice working👍

**Pull request merged**